nekoyume/Assets/_Scripts/UI/Module/GrindModule.cs

```diff
@@ -0,0 +1,324 @@
+                }
+
+                return "+0%";
+            });
+
+            MonsterCollectionStateSubject.Level
+                .Subscribe(UpdateStakingBonusObject)
+                .AddTo(gameObject);
+
+            _isInitialized = true;
+        }
+
+        public void Show(bool reverseInventoryOrder = true)
+        {
+            Initialize();
+
+            _selectedItemsForGrind.Clear();
+            grindInventory.SetGrinding(ShowItemTooltip,
+                OnUpdateInventory,
+                MakeDimConditionPredicateList(),
+                reverseInventoryOrder);
+            grindButton.Interactable = false;
+            UpdateStakingBonusObject(States.Instance.MonsterCollectionState?.Level ?? 0);
+            crystalRewardText.text = string.Empty;
+        }
+
+        private void ShowItemTooltip(InventoryItem model, RectTransform target)
+        {
+            var tooltip = ItemTooltip.Find(model.ItemBase.ItemType);
+            var isRegister = !_selectedItemsForGrind.Contains(model);
+            var isEquipment = model.ItemBase.ItemType == ItemType.Equipment;
+            var interactable =
+                isEquipment && _selectedItemsForGrind.Count < 10 && !InventoryHelper.CheckCanNotGrinding(model)
+                || !isRegister;
+            var onSubmit = isEquipment
+                ? new System.Action(() => RegisterToGrindingList(model, isRegister))
+                : null;
+            var blockMessage = model.Equipped.Value ? "ERROR_NOT_GRINDING_EQUIPPED" : "ERROR_NOT_GRINDING_10OVER";
+            var onBlock = new System.Action(() =>
+                OneLineSystem.Push(MailType.System,
+                    L10nManager.Localize(blockMessage),
+                    NotificationCell.NotificationType.Alert));
+            tooltip.Show(
+                model,
+                isRegister
+                    ? L10nManager.Localize("UI_COMBINATION_REGISTER_MATERIAL")
+                    : L10nManager.Localize("UI_COMBINATION_UNREGISTER_MATERIAL"),
+                interactable,
+                onSubmit,
+                grindInventory.ClearSelectedItem,
+                onBlock,
+                target);
+        }
+
+        private void OnUpdateInventory(Inventory inventory)
+        {
+            var selectedItemCount = _selectedItemsForGrind.Count;
+            for (int i = 0; i < selectedItemCount; i++)
+            {
+                if (inventory.TryGetModel(_selectedItemsForGrind[i].ItemBase, out var inventoryItem))
+                {
+                    inventoryItem.GrindingCount.SetValueAndForceNotify(_selectedItemsForGrind[i].GrindingCount.Value);
+                    _selectedItemsForGrind[i] = inventoryItem;
+                    itemSlots[i].UpdateSlot(_selectedItemsForGrind[i]);
+                }
+            }
+
+            grindButton.Interactable = CanGrind;
+        }
+
+        private static List<(ItemType type, Predicate<InventoryItem>)> MakeDimConditionPredicateList()
```

Comment:
```suggestion
        private static List<(ItemType type, Predicate<InventoryItem>)> DimmedFilterTypes = new List ...
```
It seems that this can be changed to field member.